exonum-java-binding/core/rust/src/proxy/node.rs

```diff
@@ -69,34 +63,34 @@ impl NodeContext {
 
     #[doc(hidden)]
     pub fn create_snapshot(&self) -> Box<Snapshot> {
-        self.blockchain.snapshot()
+        self.api_context.snapshot()
     }
 
     #[doc(hidden)]
     pub fn public_key(&self) -> PublicKey {
-        self.public_key
+        self.api_context.service_keypair().0.clone()
     }
 
     #[doc(hidden)]
-    pub fn submit(&self, transaction: RawTransaction) -> Result<Hash, failure::Error> {
-        let service_id = transaction.service_id();
-
-        if !self.blockchain.service_map().contains_key(&service_id) {
-            return Err(format_err!(
-                "Unable to broadcast transaction: service(ID={}) not found",
-                service_id
-            ));
-        }
-
-        let msg = Message::sign_transaction(
-            transaction.service_transaction(),
-            service_id,
-            self.blockchain.service_keypair.0,
-            &self.blockchain.service_keypair.1,
+    pub fn submit(&self, transaction: AnyTx) -> Result<Hash, failure::Error> {
+        // TODO: check service is active
+        let _service_id = transaction.call_info.instance_id;
+//        if !self.blockchain.service_map().contains_key(&service_id) {
+//            return Err(format_err!(
+//                "Unable to broadcast transaction: service(ID={}) not found",
+//                service_id
+//            ));
+//        }
+
+        let key_pair = self.api_context.service_keypair();
+        let verified = Verified::from_value(
```

Comment:
`signed`/`signedTx`/`signedMessage` (as we do not verify the transaction message — we create it (= sign)?

Reply:
fixed at https://github.com/exonum/exonum-java-binding/pull/1129/commits/73162f5d3143280a6bdb9ecef1b278a7808a0648

`AnyTx` -> `Signed<AnyTx>` as argument from Java